src/db/postgres.rs

```diff
@@ -39,4 +44,29 @@ impl Store for PostgresStore {
 
         Ok(row.0 == 0)
     }
+
+    async fn check_password<'a>(
+        &self,
+        user_id: &'a UserIdentifier,
+        password: &str,
+    ) -> Result<Option<Cow<'a, UserId>>, Box<dyn Error>> {
+        unimplemented!()
+    }
+
+    async fn check_otp<'a>(
```

Comment:
Similarly to above, if this is TOTP/HOTP seems like this should return the secret/encoding and let a common utility do the comparison. I am worried about side-channel attacks on all of this but doing it inside the storage makes this seem potentially more vulnerable not less.

Reply:
I believe these are regular OTPs, possibly generated by magic links sent by the server to an email address as opposed to TOTPs/HOTPs however I'm not certain